File: client-app/lib/domain/entities/chat_stream_chunk.dart
import 'package:equatable/equatable.dart';

enum ChatStreamChunkKind { text, toolStatus, notice }

class ChatStreamChunk extends Equatable {
  final ChatStreamChunkKind kind;
  final String text;
  final String? toolName;
  final int messageId;

  const ChatStreamChunk({
    required this.kind,
    required this.text,
    this.toolName,
    this.messageId = 0,
  });

  @override
  List<Object?> get props => [kind, text, toolName, messageId];
}

File: internal/usecase/chat_stream_chunk.go
package usecase

type StreamChunkKind int

const (
	StreamChunkKindText StreamChunkKind = iota
	StreamChunkKindToolStatus
	StreamChunkKindNotice
)

type ChatStreamChunk struct {
	Kind      StreamChunkKind
	Text      string
	ToolName  string
	MessageID int64
}
